coursera_autograder/commands/resources.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+
+# Copyright 2015 Coursera
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Coursera's asynchronous grader command line SDK.
+
+You may install it from source, or via pip.
+"""
+
+from coursera_autograder.commands import common
+from coursera_autograder.commands import oauth2
+import requests
+import urllib.parse
+
+"""
+target fields:
+  reservedCpu: int?
+  reservedMemory: int?
+  wallClockTimeout: int?
+"""
+
+def command_resources(args):
+    "Implements the resources subcommand"
+
+    oauth2_instance = oauth2.build_oauth2(args)
+    auth = oauth2_instance.build_authorizer()
+
+    s = requests.Session()
+    s.auth = auth
+
+    course_branch_id = args.course.replace("~", "!~") if "authoringBranch" in args.course else args.course
+    course_branch_item = '%s~%s' % (course_branch_id, args.item)
+
+    s.cookies.update({'ASG_PREFERENCE': '_VC5f2YRcjU7fHfPqB50Rrm28PLNtlnCQzx7MwMCH_6IaIhDGMc-r60p5ZRM2_sAi2yg4ZNAgs-9woRrluj2tw.4-KlTmrD1EWB_e2l8z6s1Q.WE24gP_rf77oztF1N3kO0rUjY9gnNa0ftqEazenTV3AqNQy1NeXbyWMaIUt24p_BEx0j8oGOJwXW22CL_2XOzRfqt8BeyTrUmIYFSm2_UhhjcbzRTuCx3xsb19wTJAQ9KCu2oReljTZpGADhv5XZpg'})
```

Comment:
This will be deleted once the AuthoringProgrammingAssignments API is updated.

Reply:
Which cluster this ASG is pointing to? Can you link the PR about the API update?

Reply:
I haven't opened a PR for that yet! I'll link it once I open it.

Reply:
Oh so is this pointing to your TIP? In that case I think we can hold off merging this change until the updated API is in production

Reply:
Oh definitely. I still have a lot more testing to do, but I wanted to get this initial commit out for feedback!

Reply:
Please leave it out of this PR. If needed for testing, then I think we can make the change. This repo is open-source, which is prone to accidents (including during review).